vcd/resource_vcd_vapp_nat_rules.go

```diff
@@ -0,0 +1,306 @@
+				Description:  "One of: `ipTranslation` (use IP translation), `portForwarding` (use port forwarding).",
+			},
+			"enable_ip_masquerade": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "When enabled translates a virtual machine's private, internal IP address to a public IP address for outbound traffic.",
+			},
+			"rule": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Id of the rule. Can be used to track syslog messages.",
+						},
+						"mapping_mode": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"automatic", "manual"}, false),
+							Description:  "Mapping mode. One of: `automatic`, `manual`",
+						},
+						"vm_id": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "VM to which this rule applies.",
+						},
+						"vm_nic_id": &schema.Schema{
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "VM NIC ID to which this rule applies.",
+						},
+						"external_ip": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.IsIPAddress,
+							Description:  "External IP address to forward to or External IP address to map to VM",
+						},
+						"external_port": &schema.Schema{
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "External port to forward to.",
+						},
+						"forward_to_port": &schema.Schema{
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Description: "Internal port to forward.",
+						},
+						"protocol": &schema.Schema{
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"TCP", "UDP", "TCP_UDP"}, false),
+							Description:  "Protocol to forward. One of: `TCP` (forward TCP packets), `UDP` (forward UDP packets), `TCP_UDP` (forward TCP and UDP packets).",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceVappNetworkNatRulesCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceVappNetworkNatRulesUpdate(d, meta)
+}
+
+func resourceVappNetworkNatRulesUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
+
+	vappId := d.Get("vapp_id").(string)
+	vapp, err := vdc.GetVAppById(vappId, false)
+	if err != nil {
+		return fmt.Errorf("error finding vApp. %s", err)
+	}
+	vcdClient.lockParentVappWithName(d, vapp.VApp.Name)
```

Comment:
Do you know if there is a risk of not locking Org network?

Reply:
I am not aware. I would say bigger risk is edge gtw if we think in that way. Though as we tested vApp networks are separate edge gtw's.

Reply:
Let it be then